depthcharge/transformers/analytes.py

```diff
@@ -0,0 +1,434 @@
+"""Transformer models for peptides and small molecules."""
+import torch
+
+from .. import utils
+from ..encoders import PositionalEncoder
+from ..mixins import ModelMixin, TransformerMixin
+from ..tokenizers import MoleculeTokenizer, PeptideTokenizer
+
+
+class _AnalyteTransformer(torch.nn.Module, ModelMixin, TransformerMixin):
+    """A transformer base class for peptide sequences.
+
+    Parameters
+    ----------
+    n_tokens : int or PeptideTokenizer
+        The number of tokens used to tokenize peptide sequences.
+    d_model : int
+        The latent dimensionality to represent the amino acids in a peptide
+        sequence.
+    positional_encoder : PositionalEncoder or bool
+        The positional encodings to use for the amino acid sequence. If
+        ``True``, the default positional encoder is used. ``False`` disables
+        positional encodings, typically only for ablation tests.
+    max_charge : int
+        The maximum precursor charge to embed.
+    """
+
+    def __init__(
+        self,
+        n_tokens: int | PeptideTokenizer | MoleculeTokenizer,
+        d_model: int,
+        nhead: int,
+        dim_feedforward: int,
+        n_layers: int,
+        dropout: float,
+        positional_encoder: PositionalEncoder | bool,
+    ) -> None:
+        """Initialize an AnalyteTransformer."""
+        super().__init__()
+        self._d_model = d_model
+        self._nhead = nhead
+        self._dim_feedforward = dim_feedforward
+        self._n_layers = n_layers
+        self._dropout = dropout
+
+        try:
+            n_tokens = len(n_tokens)
+        except TypeError:
+            pass
+
+        if callable(positional_encoder):
+            self.positional_encoder = positional_encoder
+        elif positional_encoder:
+            self.positional_encoder = PositionalEncoder(d_model)
+        else:
+            self.positional_encoder = torch.nn.Identity()
+
+        self.token_encoder = torch.nn.Embedding(
+            n_tokens + 1,
+            d_model,
+            padding_idx=0,
```

Comment:
**question:** Should this now use the `padding_int` that's included in the tokenizer classes?